cmd/minishift/cmd/start_preflight.go

```diff
@@ -50,13 +50,17 @@ func preflightChecksBeforeStartingHost() {
 	driverErrorMessage := "See the 'Setting Up the Driver Plug-in' topic (https://docs.openshift.org/latest/minishift/getting-started/setting-up-driver-plugin.html) for more information"
 	prerequisiteErrorMessage := "See the 'Installing Prerequisites for Minishift' topic (https://docs.openshift.org/latest/minishift/getting-started/installing.html#install-prerequisites) for more information"
 
-	preflightCheckSucceedsOrFails(
-		configCmd.SkipCheckOpenShiftRelease.Name,
-		checkOriginRelease,
-		fmt.Sprintf("Checking if requested OpenShift version '%s' is valid", viper.GetString(configCmd.OpenshiftVersion.Name)),
-		configCmd.WarnCheckOpenShiftRelease.Name,
-		fmt.Sprintf("%s is not a valid OpenShift version", viper.GetString(configCmd.OpenshiftVersion.Name)),
-	)
+	if checkInternetConnectivity() {
+		preflightCheckSucceedsOrFails(
+			configCmd.SkipCheckOpenShiftRelease.Name,
+			checkOriginRelease,
+			fmt.Sprintf("Checking if requested OpenShift version '%s' is valid", viper.GetString(configCmd.OpenshiftVersion.Name)),
+			configCmd.WarnCheckOpenShiftRelease.Name,
+			fmt.Sprintf("%s is not a valid OpenShift version", viper.GetString(configCmd.OpenshiftVersion.Name)),
+		)
+	} else {
+		fmt.Printf("-- Checking if requested OpenShift version '%s' is valid ... Skipped (offline)\n", viper.GetString(configCmd.OpenshiftVersion.Name))
```

Comment:
Sorry, one last suggestion.

I feel to align with other pre-flight check output, we should use `SKIP (offline)` instead of `Skipped (offline)`.

Reply:
I think just `SKIP` is fine. If we want to tell about the situation then it should in the debug logs IMO.  Also we do another check from inside the VM to see if we have internet connection, so no need to talk about offline bydefault. Also there is a small chance that github is not accessible but internet is accessible.